src/components/SortParam.tsx

```diff
@@ -0,0 +1,63 @@
+import classNames from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  param: keyof Person;
+};
+
+type SortParams = {
+  sort: keyof Person | null;
+  order: 'desc' | null;
+};
+
+export const SortParam: React.FC<Props> = ({ param }) => {
+  const [searchParams] = useSearchParams();
+
+  const appliedSort = searchParams.get('sort') || null;
+  const appliedOrder = searchParams.get('order') || null;
+
+  const handleSortClick = (sortType: keyof Person) => {
+    const newParams: SortParams = {
+      sort: null,
+      order: null,
+    };
+
+    if (appliedSort !== sortType) {
+      newParams.sort = sortType;
+
+      return newParams;
+    }
+
+    if (appliedSort === sortType && !appliedOrder) {
+      newParams.sort = sortType;
+      newParams.order = 'desc';
+
+      return newParams;
+    }
+
+    return newParams;
+  };
```

Comment:
The handleSortClick function does not toggle the sorting order correctly. When the appliedSort matches the sortType and appliedOrder is 'desc', the function should reset the order to null to toggle back to ascending order. Consider adding logic to handle this case.